libs/geodata/riversurvey/geodatariversurveyimporter.cpp

```diff
@@ -91,28 +37,29 @@ RivPathPoint* getPoint(double kp, std::string& kpStr, std::vector<RivPathPoint*>
 			}
 		}
 	}
-	RivPathPoint* newp = new RivPathPoint();
+	auto newp = new GeoDataRiverSurveyImporter::RivPathPoint();
+	newp->banksIsSet = false;
 	newp->realKP = kp;
 	newp->strKP = kpStr;
 	points->push_back(newp);
 	return newp;
 }
 
-void setBanks(RivPathPoint* p, double lx, double ly, double rx, double ry)
+void setBanks(GeoDataRiverSurveyImporter::RivPathPoint* p, double lx, double ly, double rx, double ry)
 {
 	p->banksIsSet = true;
-	p->leftBank.x = lx;
-	p->leftBank.y = ly;
-	p->rightBank.x = rx;
-	p->rightBank.y = ry;
+	p->leftBank.setX(lx);
+	p->leftBank.setY(ly);
+	p->rightBank.setX(rx);
+	p->rightBank.setY(ry);
 }
 
-void removeIncompleteData(std::vector<RivPathPoint*>* points)
+void removeIncompleteData(std::vector<GeoDataRiverSurveyImporter::RivPathPoint*>* points)
 {
 	int idx = 0;
 	while (idx < points->size()) {
 		auto it = points->begin() + idx;
-		RivPathPoint* p = *it;
+		auto p = *it;
 		if (! p->banksIsSet || p->altitudes.size() == 0) {
 			// remove this point;
 			points->erase(it);
```

Comment:
It looks like this is a leak.  I checked to see where it (removeIncompleteData) is was called and found out is was never called so I removed it.